api/envoy/config/route/v3/route_components.proto

```diff
@@ -1631,6 +1631,24 @@ message RateLimit {
       string default_value = 3;
     }
 
+    // The following descriptor entry is appended when the filter metadata contains a key value:
```

Comment:
Please update comments/names to be clear that what you're talking about here is the "route entry metadata". I think calling this filter metadata is a bit misleading.

Reply:
Defining which metadata is meant precisely is a common problem throughout the documentation, see: #12556 and #6156. It took me two weeks to uncover this misunderstanding. This PR is based on a suggestion from @Pchelolo .
Would a renaming from FilterMetadata to RouteEntryMetadata and adjusting the other mentions suffice?

On a side note: it would be good if the documentation have some clarification about the different metadata which can be set throughout the configuration plane. Which one available where is currently not clarified.
There is a section about [Dynamic metadata](https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata.html), but it is not clarified that metadata configured at other places will not end up there.

Reply:
Sure, the rename would be good. Thanks.

Re: docs, I agree, please file an issue or PR here.